### ISSUE TITLE
Expected output matched, but not passing the check

### ISSUE THREAD

**Issue by CurtisSalinger** [mikeizbicki/cmc-csci143#37]
Happy Saturday!

I am having a consistent problem with the SQL homework where I am getting the exact expected output, however the test cases are not passing. For instance the problem in the screenshot below is number 15, and on the left is the output of my code inside the docker-container, and the right is the `15.out` file. However, when I run `docker-compose exec pg ./check_answers.sh`, I am still getting `sql/15.sql fail`.  I thought it may be not having the `AS` column names correctly but I even went through and verified that they are all correct, so I am not sure where to go from here.

Thanks for any advice, hope everyone is having a great weekend!

<img width="333" alt="Screen Shot 2021-02-20 at 1 15 47 PM" src="https://user-images.githubusercontent.com/38367847/108604767-eda43500-737d-11eb-9f72-a6558b36b845.png">

<img width="111" alt="Screen Shot 2021-02-20 at 1 22 11 PM" src="https://user-images.githubusercontent.com/38367847/108604896-a5d1dd80-737e-11eb-91cd-e4a6ac13e9a4.png">

**Comment by mikeizbicki** [comment]
Rather than looking at the results manually, you should be using the diff command, something like:
```
$ diff expected/15.sql results/15.sql
```
The test case is a `pass` if and only if the diff command has no output.

**Comment by CurtisSalinger** [comment]
When I run that command on 15 again, I see the following which I am gathering means there is a difference in my column headers, however from this I am not sure what to change. Thanks for your help interpreting this!

<img width="660" alt="Screen Shot 2021-02-20 at 11 04 26 PM" src="https://user-images.githubusercontent.com/38367847/108615238-fcb3d300-73cf-11eb-862a-6d21491bb7ef.png">

As a heads up to anyone else who wants to run this, it is `.out` as opposed to `.sql` so for instance 15 is:
```
$ diff ./expected/15.out ./results/15.out
```

**Comment by CurtisSalinger** [comment]
Also upon further inspection, I am unsure that my `results/15.out` is updating either. I switched the code within my `sql/15.sql` file to `SELECT * FROM film` expecting to see the results of that query in `15.out` (as that is my understanding of what that file is) and it remained the results of a previous query. I may be misunderstanding this structure, but I wanted to point that out in case.

**Comment by afzal-p** [comment]
Try checking if results/15.out updates after running:

`docker-compose exec pg ./check_answers.sh`

**Comment by CurtisSalinger** [comment]
Thanks @Afzal-Patel! That helped that part of the problem, my `results` did change after running that as opposed to just running the `diff` command.

My new theory is that my results are adding spaces to fill out the columns in the text, whereas the expected are not (seen in the photo below where on the left is `expected/13.out` and on the right is my `results/13.out`). I see that on my results, all the dots after the strings represent spaces, whereas on the left, there are none. Is it part of my query that is adding those spaces? And if so does anyone know how to correct that?

Thank you for the continued help!

Updated to show the output of `$ diff ./expected/13.out ./results/13.out`

![Screen Shot 2021-02-21 at 1 22 24 PM](https://user-images.githubusercontent.com/38367847/108634514-d762ab80-7447-11eb-8010-f79f1795bcf8.png)

![Screen Shot 2021-02-21 at 2 03 36 PM](https://user-images.githubusercontent.com/38367847/108635483-99688600-744d-11eb-90af-54c8cea01d02.png)

**Comment by nguyenank** [comment]
I just looked at my `results` and `expected` outputs for that problem, and both appear to have spaces filled out on the right, so I don't think that's your issue; I think the little negation/hook thing next to in the `expected` photo is just condensing those spaces as part of your vim setup possibly, whereas they're expanded in the right screenshot? I do not know what your actual issue is, however.

**Comment by CurtisSalinger** [comment]
@nguyenank Thank you for checking for me!

**Comment by afzal-p** [comment]
I noticed the lines in vim for r/13.out are reversed 18->1. Maybe that is an issue.

**Comment by mikeizbicki** [comment]
Hmm... it looks like this is only happening for you on problems involving joins.  So I'm going to wait to try to answer this until after we talk about joins on Monday.  My best guess about what's going on here is that the issue is related to how you're doing the join.

We might need to look at this in office hours though so I can see the exact steps that you're running in order to reproduce this on my end.

**Comment by CurtisSalinger** [comment]
Thank you, I will come to your office hours today to look this over.

**Comment by mikeizbicki** [comment]
We've solved the issue problem!

**For everyone else:** The source of the problem is that you've actually modified all the files in the `expected` folder away from what they are supposed to be.  For @CurtisSalinger , this was caused by his `.vimrc` file having lines in it that automatically remove whitespace from a file that he opens.  (Something like described in https://vim.fandom.com/wiki/Remove_unwanted_spaces).  I personally used to have this enabled in my own vimrc, but I have disabled it because it can really bork git repos.

To fix the problem, you need to replace the output files in your `expected` folder with their original contents.  The following series of commands will do this:
```
$ git remote add upstream https://github.com/mikeizbicki/pagila-hw 
$ git fetch upstream master
$ git checkout upstream/master
$ cp -r expected expected_original
$ git checkout master
$ rm -rf expected
$ mv expected_original expected
$ git add expected
$ git commit -m 'reset test cases'
```

**Comment by CurtisSalinger** [comment]
So close to being able to close this issue! Now that I have committed and am getting all passes, I am still running into a continuous integration error shown below.

I am not sure where this error is coming from (though online I saw something about an issue using SIGTERM?) so I am unsure what to do to fix it.

Thanks again for helping me through this today!

<img width="763" alt="Screen Shot 2021-02-22 at 8 21 30 PM" src="https://user-images.githubusercontent.com/38367847/108791017-1712bb80-754c-11eb-9857-cdb08ddb68af.png">

**Comment by nguyenank** [comment]
I think the process completed is just the standard completion thing; the issue is that you still aren't passing `sql/09.sql`.

**Comment by CurtisSalinger** [comment]
Update: Without changing anything and running `docker-compose exec pg ./check_answers.sh`, I see all pass. Going into my `results/09.out` I see it is adding an extra empty line at the bottom of the file (A). I confirmed this by manually removing the extra line, running the `diff` command, seeing no output, running `check_answers.sh`, seeing all passes, and then running `diff` again and seeing the line appended. So when I push this to GitHub, it doesn't pass the workflows test, but it passes my local `check_answers.sh` (B).

I checked all my other results and this is the only one that the extra line is being added for even though my SQL does not seem to be abnormal (C). Sorry to have so many questions on this one issue, I am just at a loss for what to do here!

A.
<img width="568" alt="Screen Shot 2021-02-22 at 8 54 28 PM" src="https://user-images.githubusercontent.com/38367847/108792814-298ef400-7550-11eb-9b86-c413fc600bca.png">

B.
<img width="679" alt="Screen Shot 2021-02-22 at 8 57 53 PM" src="https://user-images.githubusercontent.com/38367847/108793049-a621d280-7550-11eb-9727-2fd88be0a547.png">

C.
<img width="700" alt="Screen Shot 2021-02-22 at 9 00 05 PM" src="https://user-images.githubusercontent.com/38367847/108793189-f7ca5d00-7550-11eb-9bf7-40b5e166aa67.png">
